src/routes/users.js

```diff
@@ -1,271 +1,324 @@
 import express from "express";
 import * as auth0_client from "../utils/auth0_client";
-import { PrismaClient } from '@prisma/client';
+import { PrismaClient } from "@prisma/client";
 import { checkJwt } from "../auth";
 import { sendWebhookMessage } from "../utils/webhook_client";
 
 const router = express.Router();
 const prisma = new PrismaClient();
 
-
 // API version: 2.0
-router.get('/:id', checkJwt, async (req, res) => {
+router.get("/:id", checkJwt, async (req, res) => {
   const user_id = req.params.id;
   const token_sub = req.user.sub;
-  if(token_sub !== user_id) {
-    res.status(403).send({course_table: null, message: "you are not authorized to get this user data."});
+  if (token_sub !== user_id) {
+    res.status(403).send({
+      course_table: null,
+      message: "you are not authorized to get this user data.",
+    });
     return;
   }
   try {
     const token = await auth0_client.get_token();
     const auth0_user = await auth0_client.get_user_by_id(user_id, token);
-    if (!auth0_user){
-      res.status(404).send({user: null, message: "User is not registered in Auth0"});
+    if (!auth0_user) {
+      res
+        .status(404)
+        .send({ user: null, message: "User is not registered in Auth0" });
       return;
     }
-    let db_user = await prisma.users.findUnique({ 
+    let db_user = await prisma.users.findUnique({
       where: { id: user_id },
       include: {
         major_dept: true,
         d_major_dept: true,
-      }
+      },
     });
-    if(db_user){
+    if (db_user) {
       db_user = await process_user_info(db_user);
-      res.status(200).send({user: {db: db_user, auth0: auth0_user}, message: "Successfully get user by id."});
-    }else{
-      res.status(400).send({user: null, message: "User not found in DB."});
+      res.status(200).send({
+        user: { db: db_user, auth0: auth0_user },
+        message: "Successfully get user by id.",
+      });
+    } else {
+      res.status(200).send({ user: null, message: "User not found in DB." });
```

Comment:
Bug 3: we should return 200 instead of 400 ( in `v1`, we return 200). Because when user not found in database, we  would want to immediately trigger `register` user api. (instead of crashing frontend)